x-pack/plugins/cases/server/services/user_actions/index.ts

```diff
@@ -378,6 +380,58 @@ export class CaseUserActionService {
       throw error;
     }
   }
+
+  public async findStatusChanges({
+    unsecuredSavedObjectsClient,
+    caseId,
+    filter,
+  }: {
+    unsecuredSavedObjectsClient: SavedObjectsClientContract;
+    caseId: string;
+    filter?: KueryNode;
+  }): Promise<Array<SavedObject<CaseUserActionResponse>>> {
+    try {
+      this.log.debug('Attempting to find status changes');
+
+      const updateActionFilter = buildFilter({
+        filters: Actions.update,
+        field: 'action',
+        operator: 'or',
+        type: CASE_USER_ACTION_SAVED_OBJECT,
+      });
+
+      const statusChangeFilter = buildFilter({
+        filters: ActionTypes.status,
+        field: 'type',
+        operator: 'or',
+        type: CASE_USER_ACTION_SAVED_OBJECT,
+      });
+
+      const combinedFilters = combineFilters([updateActionFilter, statusChangeFilter, filter]);
+
+      const finder =
+        unsecuredSavedObjectsClient.createPointInTimeFinder<CaseUserActionAttributesWithoutConnectorId>(
+          {
+            type: CASE_USER_ACTION_SAVED_OBJECT,
+            hasReference: { type: CASE_SAVED_OBJECT, id: caseId },
+            sortField: defaultSortField,
+            sortOrder: 'asc',
+            filter: combinedFilters,
+            perPage: MAX_DOCS_PER_PAGE,
+          }
+        );
+
+      let userActions: Array<SavedObject<CaseUserActionAttributesWithoutConnectorId>> = [];
+      for await (const findResults of finder.find()) {
+        userActions = userActions.concat(findResults.saved_objects);
```

Comment:
optimization nit: What do you think about doing the transformation on the fly inside the for loop instead of iterating the user actions again after the loop?